src/Agent.Plugins/Artifact/PipelineArtifactProvider.cs

```diff
@@ -48,14 +30,12 @@ public PipelineArtifactProvider(AgentTaskPluginExecutionContext context, VssConn
             this.context = context;
             this.connection = connection;
             dedupStoreHttpClient.SetTracer(tracer);
```

Comment:
I think you should also put this line into the factory where we create the client, and remove the line above for dedupStoreHttpClient initialization

Reply:
lol... i guess this is another example of code going nowhere.  Good catch.

Reply:
The other code sets tracer on the client properly but we can delete these lines because they go nowhere.